sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -40,12 +40,20 @@
 class SdkHarness(object):
   REQUEST_METHOD_PREFIX = '_request_'
 
-  def __init__(self, control_address, worker_count):
+  def __init__(self, control_address, worker_count,
+               control_channel=None, credentials=None):
     self._worker_count = worker_count
     self._worker_index = 0
+    if control_channel is None:
+      logging.info('Creating insecure channel.')
+      self._control_channel = grpc.insecure_channel(control_address)
+    else:
+      logging.info('Using provided channel.')
+      self._control_channel = control_channel
```

Comment:
Why are we passing in a pre-created channel in the secure case rather than creating it here given the address and credentials? Seems a bit inconsistent.

Reply:
Updated, PTAL.